source/val/validation_state.cpp

```diff
@@ -332,6 +332,24 @@ void ValidationState_t::RegisterCapability(SpvCapability cap) {
     desc->capabilities.ForEach(
         [this](SpvCapability c) { RegisterCapability(c); });
   }
+
+  switch (cap) {
+    case SpvCapabilityInt16:
+      features_.declare_int16_type = true;
+      break;
+    case SpvCapabilityFloat16:
+    case SpvCapabilityFloat16Buffer:
+      features_.declare_float16_type = true;
+      break;
+    case SpvCapabilityStorageUniformBufferBlock16:
+    case SpvCapabilityStorageUniform16:
+    case SpvCapabilityStoragePushConstant16:
+    case SpvCapabilityStorageInputOutput16:
+      features_.declare_int16_type = true;
```

Comment:
Correct, Int16 also allows arithmetic operations on those types.  But the validator doesn't check that yet.  Also, that's why the feature is called just *declare* 16bit types.   Later we will add something like int16_arithmetic, and that will distinguish the cases.